component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java

```diff
@@ -685,7 +685,17 @@ public int getIdentitiesByFirstCharacterOfNameCount(String providerId, ProfileFi
     ExtendProfileFilter xFilter = new ExtendProfileFilter(profileFilter);
     xFilter.setIdentityIds(relations);
     ListAccess<IdentityEntity> list = getIdentityDAO().findIdentities(xFilter);
-    return EntityConverterUtils.convertToIdentities(list, offset, limit);
+    List<Identity> identitiesList = EntityConverterUtils.convertToIdentities(list, offset, limit);
+    if (profileFilter != null && profileFilter.getSorting() != null
+        && Sorting.SortBy.TITLE.equals(profileFilter.getSorting().sortBy)) {
+      return sortIdentities(identitiesList, Profile.FULL_NAME);
```

Comment:
The sort is done after loading the current page of identities (offset to limit), so it is done only on this subset of identities. It means if we have more identities than this subset, we can have wrong sort when the user will load several pages of identities, no ?

Reply:
after checking with 4.4.x version it's not a real lazy loading it will be always called with convertToIdentities(list, 0, 0) so it will load all the list from the sort param
selectQuery.setMaxResults(Integer.MAX_VALUE);

Reply:
so no we will have only one subset of identities so always  the same sort .

Reply:
here the enty point from my debugger:
      https://github.com/exoplatform/social/blob/064e04b6707d04cd5359107ef38bdb3ad3f1703e/component/core/src/main/java/org/exoplatform/social/core/identity/SpaceMemberFilterListAccess.java#L103 so limit = offset=0 

https://github.com/exoplatform/social/blob/stable/4.4.x/component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java#L229

Reply:
If it loads all the users of the platform, we will face performance issues.

Reply:
I did a test, the method getSpaceMemberIdentitiesByProfileFilter is called several times. Some of these calls are done with offset=limit=0 to count the total number of members, but the call to load the members of a page uses offset and limit not always equal to 0 (first call : offset=0 and limit=45). So the sort will be done only page by page and will not fix the customer issue.